### ISSUE TITLE
Suggested enhancements to CheckProjects

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/coreclr#19166]
The CheckProjects jit test is intended to verify that all the jit test project files meet our expectations:

* test suffix convention (_d, _do, etc) matches projects settings
* the IL generated for the test does not depend on the ambient build environment

#19132 points out some example where the optimization settings were not checked
#19131 some examples where the suffix convention was misused

We should enhance CheckProjects to catch those cases.

#19165 is removing check exclusions for the codegen bring up tests, so all jit tests (modulo the bugs above) should now conform.

In addition because it can rewrite project files, we might want to enhance it so that all projects have a unique GUID.

If / when we convert tests over to using SDK style projects we should make corresponding updates to the tool.

**Comment by mazong1123** [comment]
Is this issue still valid? If yes I can help on this. Seems not a big effort and a good start to understand the jit compiler :)

**Comment by mazong1123** [comment]
I found some *.ilproj has multiple `<Optimize>` Properties. For example in `tests\src\JIT\Methodical\nonvirtualcall\generics2_r.ilproj`, there are:
```
  <PropertyGroup>
    <DebugType>PdbOnly</DebugType>
    <Optimize>False</Optimize>
  </PropertyGroup>
```
```
  <PropertyGroup>
    <Optimize>True</Optimize>
  </PropertyGroup>
```

To align `DebugType`'s logic, this falls to `Odd` category as the 2 `Optimize` property has different values. In my opinion we should change the value of second `Optimize` to `false` to align to the `_r` suffix of the file name. Another option is to delete the second `Optimize` as it's redundancy. Which one sounds more compliant?

**Comment by mazong1123** [comment]
I'd split this work item to following small ones:

- [ ] Check optimization settings and output the project files need to be fixed.
- [ ] Add logic to auto-fix the optimization settings.
- [ ] Check suffix misuse and output the project files need to be fixed.
- [ ] Add logic to auto-fix the suffix misuse.
- [ ] Ensure all projects have an unique GUID.

I'll send separate PR to complete each of the tasks.

**Comment by mikedn** [comment]
> In addition because it can rewrite project files, we might want to enhance it so that all projects have a unique GUID.

@AndyAyersMS Does anyone actually need that? The GUID (and a bunch of other stuff) is normally just unnecessary VS generated clutter.

**Comment by AndyAyersMS** [comment]
@mazong1123 yes this is still relevant, though at some point we will probably transition over to using SDK style projects so there may be limited value in a full cleanup.

So I'd focus on the first 4 steps, all of which are remain relevant even with new projects and all of which impact our test coverage, and not worry about the GUID.

**Comment by BruceForstall** [comment]
@echesakovMSFT Did you look in to CheckProjects when you converted tests to SDK style? (i.e., are any changes needed there?)

**Comment by echesakovMSFT** [comment]
@BruceForstall I have not but it looks that we can simplify CheckProjects. For example, I don't think we would need this https://github.com/dotnet/coreclr/blob/d8b7989d76cf239586abbc5c604af245e2b2fc1f/tests/src/JIT/CheckProjects/CheckProjects.cs#L140-L149
